src/store.rs

```diff
@@ -136,6 +138,19 @@ impl CertStore {
         unsafe { self.do_find(CERT_FIND_HASH, &hash_blob as *const _ as _) }
     }
 
+    /// Find list of certificates matching the SHA256 hash
+    pub fn find_by_sha256<D>(&self, hash: D) -> Result<Vec<CertContext>>
+    where
+        D: AsRef<[u8]>,
+    {
+        let hash_blob = CRYPT_INTEGER_BLOB {
+            cbData: hash.as_ref().len() as u32,
+            pbData: hash.as_ref().as_ptr() as _,
+        };
+        unsafe { self.cert_find_by_sha256(CERT_FIND_ANY, &hash_blob as *const _ as _) }
```

Comment:
>cert_find_by_sha256 [](http://example.com/codeflow?start=22&length=19)

Add comments about:

On later OS releases, we added CERT_FIND_SHA256_HASH. However, rustls-cng could be installed on earlier OS release where this FIND_SHA256 isn't present.

However, the CERT_SHA256_HASH_PROP_ID is present. So will need to add a new internal find function that gets and compares the SHA256 property.

Reply:
done